### PR TITLE
RateLimiter: Added Rate limiting to Submission and User COntroller

### PR DESCRIPTION
**Pull request by amankumarkeshu**
Added Rate limiting to Submission and User COntroller
Currently only 5 requests can be made in an interval of 4 seconds.

### CODE REVIEW

File: src/main/java/com/projects/onlinejudge/controller/PingController.java [hunk 1]
```diff
@@ -1,12 +1,28 @@
...
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RestController;
+import io.github.resilience4j.bulkhead.annotation.Bulkhead;
+import io.github.resilience4j.ratelimiter.RequestNotPermitted;
+import io.github.resilience4j.ratelimiter.annotation.RateLimiter;
+import io.github.resilience4j.retry.annotation.Retry;
 
 @RestController
 public class PingController {
+    private static final String ORDER_SERVICE ="orderService" ;
```
Comment (akash13s) [inline src/main/java/com/projects/onlinejudge/controller/PingController.java:16]: Please rename it to 'SUBMISSION_SERVICE'

File: src/main/java/com/projects/onlinejudge/controller/SubmissionController.java [hunk 2]
```diff
@@ -7,16 +7,20 @@
...
 import org.springframework.web.bind.annotation.*;
 import org.springframework.web.multipart.MultipartFile;
+import io.github.resilience4j.ratelimiter.annotation.RateLimiter;
 
 
 @RestController
 @RequestMapping("/api/submissions")
 public class SubmissionController {
 
+    private static final String ORDER_SERVICE ="orderService" ;
```
Comment (akash13s) [inline src/main/java/com/projects/onlinejudge/controller/SubmissionController.java:17]: Please rename it to 'SUBMISSION_SERVICE'

**Review by akash13s — COMMENTED** [review]
Please rename the variable in similar places. Apart from that all other things look good.

**Comment by amankumarkeshu** [comment]
@Akash Shrivastva ***@***.***> We are using single config
for all our APIs if we rename it we would have to add multiple configs in
applications.yml for each service. Let me know if that looks good to you.

**Comment by amankumarkeshu** [comment]
I can rename the variable name. But still order service would be the value
assigned to submission service.

**Comment by akash13s** [comment]
Okay @amankumarkeshu let it be as it is for now. LGTM.

**Pull request merged**